src/main/java/org/folio/service/caches/MappingParametersCache.java

```diff
@@ -70,64 +108,87 @@ public MappingParametersCache(Vertx vertx, RestClient restClient) {
    */
   public Future<MappingParameters> get(OkapiConnectionParams params) {
     try {
-      return Future.fromCompletionStage(cache.get(params.getTenantId(), (key, executor) -> loadMappingParameters(params)));
+      return Future.fromCompletionStage(cache.get(params.getTenantId(), (key, executor) ->
+        loadMappingParameters(params).toCompletionStage().toCompletableFuture()));
     } catch (Exception e) {
       LOGGER.warn("get:: Error loading organizations from cache, tenantId: '{}'", params.getTenantId(), e);
       return Future.failedFuture(e);
     }
   }
 
   /**
-   * Generates {@link MappingParameters} with collection of {@link org.folio.Organization}
+   * Generates {@link MappingParameters}
    *
    * @param params {@link OkapiConnectionParams} connection params
    * @return CompletableFuture with {@link MappingParameters} object
    */
-  private CompletableFuture<MappingParameters> loadMappingParameters(OkapiConnectionParams params) {
+  private Future<MappingParameters> loadMappingParameters(OkapiConnectionParams params) {
+    String tenantId = params.getTenantId();
+    LOGGER.debug("loadMappingParameters:: Trying to load mapping parameters '{}' for cache, okapi url: {}, tenantId: {}",
+      tenantId, params.getOkapiUrl(), params.getTenantId());
+
+    Future<List<Location>> locationsFuture = getLocations(params);
+    Future<List<Mtype>> materialTypesFuture = getMaterialTypes(params);
+    Future<List<ContributorNameType>> contributorNameTypesFuture = getContributorNameTypes(params);
+    Future<List<Organization>> organizationsFuture = getOrganizations(params);
+    Future<List<Fund>> fundsFuture = getFunds(params);
+    Future<List<ExpenseClass>> expenseClassesFuture = getExpenseClasses(params);
+    Future<List<IdentifierType>> identifierTypesFuture = getIdentifierTypes(params);
+    Future<List<AcquisitionsUnit>> acquisitionsUnitsFuture = getAcquisitionsUnits(params);
+    Future<List<AcquisitionMethod>> acquisitionsMethodsFuture = getAcquisitionMethods(params);
+    Future<List<String>> tenantConfigurationAddressesFuture = getTenantConfigurationAddresses(params);
```

Comment:
seems knowing exacty type of variable is not much important here, as a option we can use var it make it this block of code more easier to read
```

   var locationsFuture = getLocations(params);
    var materialTypesFuture = getMaterialTypes(params);
    var contributorNameTypesFuture = getContributorNameTypes(params);
    var organizationsFuture = getOrganizations(params);
    var fundsFuture = getFunds(params);
    var expenseClassesFuture = getExpenseClasses(params);
    var identifierTypesFuture = getIdentifierTypes(params);
    var acquisitionsUnitsFuture = getAcquisitionsUnits(params);
    var acquisitionsMethodsFuture = getAcquisitionMethods(params);
    var tenantConfigurationAddressesFuture = getTenantConfigurationAddresses(params);
```

Reply:
Done, thank you